python/ray/data/_internal/logical/rules/randomize_blocks.py

```diff
@@ -0,0 +1,69 @@
+from collections import deque
+
+from ray.data._internal.logical.interfaces import LogicalOperator, LogicalPlan, Rule
+from ray.data._internal.logical.operators.all_to_all_operator import (
+    AbstractAllToAll,
+    RandomizeBlocks,
+)
+
+
+class ReorderRandomizeBlocksRule(Rule):
+    """Rule for reordering RandomizeBlocks logical operator.
+
+    Reordering RandomizeBlocks operators is to help fuse multiple AbstractMap operators together for better performance.
+
+    1. Dedupes multiple RandomizeBlocks operators if they are not seeded.
+    2. Moves RandomizeBlocks operator to the end of a sequence of AbstractMap
+    operators. RandomizeBlocks operators are not moved across AbstractAllToAll operator
+    boundaries.
+    """
+
+    def apply(self, plan: LogicalPlan) -> LogicalPlan:
+        optimized_dag: LogicalOperator = self._apply(plan.dag)
+        return LogicalPlan(dag=optimized_dag)
+
+    def _apply(self, op: LogicalOperator) -> LogicalOperator:
+        seeds = []
```

Comment:
as discussed, let's keep a list of `RandomizeBlocks` operator, instead of `seed`, to be future-proof when we add more attributes into `RandomizeBlocks`. Thanks.

Reply:
updated